docs/screenshots.md

```diff
@@ -0,0 +1,99 @@
+Auto generate screenshots for user manual
+=========================================
+
+[Github Issue Page](https://github.com/learningequality/ka-lite/issues/2237)
+
+
+## General Approaches
+
+These screenshots will be used for the user manual.  A screenshot may be used directly, sliced, cropped, merged, or annotated.  It can also be used more than once on the manual.  So the most viable solution is to have a list of screenshots to be generated by the script.
+
+We create a management command to be ran as `python manage.py screenshots` and use Selenium to take screenshot of the web browser: [Take a screenshot with Selenium WebDriver](http://stackoverflow.com/a/6282203/845481).  It uses the `firefox` webdriver by default.
+
+This command uses a Screenshot class which takes "orders" from a `screenshots.json` on what, how, and when to do the screenshots.
+
+The `screenshots.json` data file "drives" the Screenshot class.  Each entry has a starting url and can optionally have actions that can be done when the Selenium browser loads the page.  Actions include filling-in form fields, submitting the form, or taking a screenshot in between the actions.  So an entry therefore may have one or more screenshots:
+    
+    1. based on the `slug` key;
+    2. based on each `<slug>` key of the `inputs` array.  It is an array to make sure that the actions are parsed and sorted.
+
+The screenshot filename value is taken from the value of the above-mentioned keys which must be unique.
+
+The command creates a new sqlite3 database for this purpose and deletes it afterward and is specified as `SCREENSHOTS_DATABASE_NAME` on `/kalite/distributed/settings.py` among with other settings for the command.
+
+
+## Requirements
+1. Location to save the screenshots which is `/data/screenshots/` folder by default.
+1. Browser to use for the Selenium webdriver, dimensions, and versions?  Defaults to Firefox v31.0 with 1024x68 window size
+1. Operating systems and versions?
```

Comment:
Indeed, I wouldn't necessarily expect this to run on an arbitrary machine, but since it's only meant to be used by devs for building docs that should be okay, right?